sklearn/manifold/tests/test_t_sne.py

```diff
@@ -233,6 +278,87 @@ def test_pca_initialization_not_compatible_with_precomputed_kernel():
                          tsne.fit_transform, np.array([[0.0], [1.0]]))
 
 
+def test_answer_gradient_two_particles():
+    """ This case with two particles test the tree with only a single
+        set of children.
+
+        These tests & answers have been checked against
+        the reference implementation by LvdM
+    """
+    pos_input = np.array([[1.0, 0.0], [0.0, 1.0]])
+    pos_output = np.array([[-4.961291e-05, -1.072243e-04],
+                           [9.259460e-05, 2.702024e-04]])
+    neighbors = np.array([[1],
+                          [0]])
+    grad_output = np.array([[-2.37012478e-05, -6.29044398e-05],
+                            [2.37012478e-05, 6.29044398e-05]])
+    yield _run_answer_test, pos_input, pos_output, neighbors, grad_output
```

Comment:
I think this is more difficult to understand than `_run_answer_test(pos_input, pos_output, neighbors, grad_output)`